addons/sale/sale.py

```diff
@@ -886,12 +886,18 @@ def product_id_change(self, cr, uid, ids, pricelist, product, qty=0,
                     uos = False
             else:
                 uos = False
-        fpos = fiscal_position and self.pool.get('account.fiscal.position').browse(cr, uid, fiscal_position) or False
+        fpos = fiscal_position and self.pool.get(
+                'account.fiscal.position'
+            ).browse(cr, uid, fiscal_position) or False
         if update_tax: #The quantity only have changed
-            result['tax_id'] = self.pool.get('account.fiscal.position').map_tax(cr, uid, fpos, product_obj.taxes_id)
+            result['tax_id'] = self.pool.get(
+                'account.fiscal.position'
+            ).map_tax(cr, uid, fpos, product_obj.taxes_id)
 
         if not flag:
-            result['name'] = self.pool.get('product.product').name_get(cr, uid, [product_obj.id], context=context_partner)[0][1]
+            result['name'] = product_obj.partner_ref
```

Comment:
How is name_get wrong? If I look at the definition of name_get in product/product.py, it does attempt to retrieve the partner specific name and product code, just like the function that retrieves the value for partner_ref.